# Módulo 10 tarefa 2

Considere a mesma base de dados utilizada na tarefa 1 ```previsao_de_renda.csv```.

1) Com o que você viu em aula, faça uma análise das variáveis qualitativas (posse de imóvel e posse de veículo) para verificar se são boas preditoras de *renda*. Alguma delas parece discriminar *renda* de forma estatisticamente significante?

In [1]:
import pandas as pd
import scipy.stats as stats

In [2]:
arquivo = 'previsao_de_renda.csv'
dados = pd.read_csv(arquivo)

In [3]:
# Verificando as variáveis qualitativas (Ver a distribuição de quem tem ou não tem imóvel e veículo)
print(dados['posse_de_imovel'].value_counts())
print(dados['posse_de_veiculo'].value_counts())

posse_de_imovel
True     10082
False     4918
Name: count, dtype: int64
posse_de_veiculo
False    9167
True     5833
Name: count, dtype: int64


In [4]:
# Comparando as médias de renda entre quem tem e quem não tem imóvel
media_renda_imovel = dados.groupby('posse_de_imovel')['renda'].mean()
media_renda_veiculo = dados.groupby('posse_de_veiculo')['renda'].mean()

print("\nMédia de renda por posse de imóvel:")
print(media_renda_imovel)

print("\nMédia de renda por posse de veículo:")
print(media_renda_veiculo)


Média de renda por posse de imóvel:
posse_de_imovel
False    4426.970482
True     4721.052704
Name: renda, dtype: float64

Média de renda por posse de veículo:
posse_de_veiculo
False    4390.506963
True     4992.579609
Name: renda, dtype: float64


In [5]:
# Realizando um teste t para ver se as diferenças são significativas
t_test_imovel = stats.ttest_ind(dados[dados['posse_de_imovel'] == 1]['renda'],
                                dados[dados['posse_de_imovel'] == 0]['renda'])

t_test_veiculo = stats.ttest_ind(dados[dados['posse_de_veiculo'] == 1]['renda'],
                                 dados[dados['posse_de_veiculo'] == 0]['renda'])

print("\nTeste t para posse de imóvel:", t_test_imovel)
print("Teste t para posse de veículo:", t_test_veiculo)


Teste t para posse de imóvel: TtestResult(statistic=3.6544544037866884, pvalue=0.0002586087928865693, df=14998.0)
Teste t para posse de veículo: TtestResult(statistic=7.781733886565539, pvalue=7.618601898603797e-15, df=14998.0)


**Distribuição de quem tem e quem não tem imóvel e veículo:**

**Posse de imóvel:**

Resultado do teste T:
𝑡
=
3.65
t=3.65,
𝑝
=
0.00026
p=0.00026

Isso significa que a diferença entre a renda de quem tem e quem não tem imóvel é estatisticamente significativa.

**Posse de veículo:**

Resultado do teste T:
𝑡
=
7.78
t=7.78,
𝑝
<
0.00001
p<0.00001

A diferença entre a renda de quem tem e quem não tem veículo é ainda mais significativa.

**Conclusão:** Tanto a posse de imóvel quanto a posse de veículo são boas preditoras de renda, mas a posse de veículo parece discriminar a renda de forma mais forte, com uma diferença de renda maior e um resultado estatisticamente mais significativo.